src/app/clusters/descriptor/descriptor.h

```diff
@@ -0,0 +1,27 @@
+/**
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace Descriptor {
+
+inline constexpr uint16_t kClusterRevision = 2;
```

Comment:
Why in a header?it's not clear why there would be any importance for an API user to now this? Are there more details to justify? I would prefer a method on the descriptor itself if it's that important for callers to know.

Reply:
I'm going to have objects representing cluster instances in Matter.framework.  Mostly these will be API-consumer-provided, and the clusterRevision (and implementation) will be provided by the consumer.  But for Descriptor I will be using the SDK implementation, and the consumer will not be able to set the revision reported by those objects; it should just be set to whatever revision the SDK actually implements.

Now I could set it to 0 or something instead, but that would be pretty confusing to API consumers.

I could also just hardcode the revision in Matter.framework and add a CI test that the two hardcoded numbers match, but it seems better to share a single source of truth instead.

I'm not sure what you mean by "a method on the descriptor itself".  There is no "descriptor itself" right now that is in any way accessible to anyone, right?  All it has is the AttributeAccessInterface it registers and no API at all.

Reply:
That is right: there is no public API to most clusters, which is the problem. I will approve this PR, but it would be important to consider avoiding too many floating constants for clusters, when the cluster implementation itself could have given the value...